osu.Game/Overlays/Toolbar/ToolbarButton.cs

```diff
@@ -157,25 +157,27 @@ protected ToolbarButton()
             };
         }
 
-        private readonly Cached tooltipKeyBinding = new Cached();
+        private RealmKeyBinding realmKeyBinding;
 
-        [BackgroundDependencyLoader]
-        private void load()
+        protected override void LoadComplete()
         {
-            keyBindings.KeyBindingChanged += () => tooltipKeyBinding.Invalidate();
-            updateKeyBindingTooltip();
-        }
+            base.LoadComplete();
 
-        private void updateKeyBindingTooltip()
-        {
-            if (tooltipKeyBinding.IsValid)
-                return;
+            if (Hotkey != null)
+            {
+                realmKeyBinding = realmFactory.Context.All<RealmKeyBinding>().FirstOrDefault(rkb => rkb.RulesetID == null && rkb.ActionInt == (int)Hotkey.Value);
 
-            var binding = keyBindings.Query().Find(b => (GlobalAction)b.Action == Hotkey);
-            var keyBindingString = binding?.KeyCombination.ReadableString();
-            keyBindingTooltip.Text = !string.IsNullOrEmpty(keyBindingString) ? $" ({keyBindingString})" : string.Empty;
+                if (realmKeyBinding != null)
+                {
+                    realmKeyBinding.PropertyChanged += (sender, args) =>
+                    {
+                        if (args.PropertyName == nameof(realmKeyBinding.KeyCombinationString))
+                            updateKeyBindingTooltip();
+                    };
```

Comment:
this is also pretty raw (brings me back to WPF days a little bit...) - another candidate for abstracting around a little bit if we end up using this often

Reply:
For sure. In this particular case we don't even need to check the property name; I just included it to show the raw realm flow here really.